### ISSUE TITLE
StreamDecoratorTrait::__toString could cause fatal errors with trigger_error

### ISSUE THREAD

**Issue by origaminal** [guzzle/psr7#180]
There is the work around in `StreamDecoratorTrait::__toString` method for inability of `__toString` to throw exceptions.
```php
public function __toString()
    {
        try {
            if ($this->isSeekable()) {
                $this->seek(0);
            }
            return $this->getContents();
        } catch (\Exception $e) {
            // Really, PHP? https://bugs.php.net/bug.php?id=53648
            trigger_error('StreamDecorator::__toString exception: '
                . (string) $e, E_USER_ERROR);
            return '';
        }
    }
```
The problem that there could be user-defined handlers that throws Exceptions on their own like it was in 2.7 version of Symfony

https://github.com/symfony/symfony/blob/ff630e19bbdea66d972a969445ecc2396c754aac/src/Symfony/Component/Debug/ErrorHandler.php#L399-L415

This leads to the `Fatal Error` which is quite tricky to debug. 

Whether there could be any better strategy for this piece of code? Maybe to wrap `trigger_error` in a `try-catch` block and do `return ''` if an exception was thrown? Or at least to add the comment like `if you still do get "FatalError ::__toString must not throw an exception" it is cause your error handler throw an exception` - this could save some time in debugging.

**Comment by sagikazarmark** [comment]
Is the fatal error saying that `__toString` should not throw exceptions? Can you please provide more information (code, error messages, etc), what you exactly see? Thanks!

**Comment by Tobion** [comment]
Closing as the problem will be solved for upcoming php versions #271
